dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -405,4 +412,57 @@ public static long countFailures(List<ArtifactResults> table, String columnName)
         .filter(row -> row.getResult(columnName) == null || row.getFailureCount(columnName) > 0)
         .count();
   }
+
+  private static final int SUMMARIZING_DEPENDENCY_PATH_SIZE_THRESHOLD = 5;
+
+  /**
+   * Returns mapping from jar files to summaries of the root problem in their {@link
+   * DependencyPath}s. The summary explains common patterns ({@code groupId:artifactId}) in the path
+   * elements. The returned map does not have a key for a jar file when the length of its {@link
+   * DependencyPath} list is smaller than {@link #SUMMARIZING_DEPENDENCY_PATH_SIZE_THRESHOLD} or a
+   * common pattern is not found among the elements in the paths.
+   *
+   * <p>Using this summary in the BOM dashboard avoids repetitive items in the {@link
+   * DependencyPath} list that share the same root problem caused by widely-used libraries, for
+   * example, {@code commons-logging:commons-logging}, {@code
+   * com.google.http-client:google-http-client} and {@code log4j:log4j}.
+   */
+  private static ImmutableMap<String, String> jarToDependencyPathSummary(
+      ListMultimap<Path, DependencyPath> jarToDependencyPaths) {
+    // Freemarker is not good at handling non-string key. Path object in .ftl is automatically
+    // converted to String. https://freemarker.apache.org/docs/app_faq.html#faq_nonstring_keys
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+
+    for (Path jar : jarToDependencyPaths.keySet()) {
+      // To keep order of the common artifact when shown in dashboard
+      LinkedHashSet<String> versionlessCoordinates = null; // null for 1st iteration
+      boolean firstIteration = true;
+      List<DependencyPath> dependencyPaths = jarToDependencyPaths.get(jar);
+      for (DependencyPath dependencyPath : dependencyPaths) {
+        // List of versionless coordinates ("groupId:artifactId")
+        ImmutableList<String> versionlessCoordinatesInPath =
+            dependencyPath.getPath().stream().map(Artifacts::makeKey).collect(toImmutableList());
+        if (firstIteration) {
+          firstIteration = false;
+          versionlessCoordinates = Sets.newLinkedHashSet(versionlessCoordinatesInPath);
+        } else {
+          // intersection of elements in DependencyPaths
+          versionlessCoordinates.retainAll(versionlessCoordinatesInPath);
+        }
+      }
+
+      // When dependencyPaths is not empty, versionlessCoordinates is not null
+      if (dependencyPaths.size() > SUMMARIZING_DEPENDENCY_PATH_SIZE_THRESHOLD
+          && versionlessCoordinates.size() > 1) { // The last elements are always same among paths
+        StringBuilder messageBuilder = new StringBuilder();
+        messageBuilder.append("There are " + dependencyPaths.size() + " paths to the artifact. ");
+        messageBuilder.append("All of them have the same artifacts: ");
+        messageBuilder.append(Joiner.on(" > ").join(versionlessCoordinates));
```

Comment:
This is the information we care about. It should be first.

Reply:
Updated. For example it now has 

> Artifacts 'com.google.cloud:google-cloud-core-http > com.google.http-client:google-http-client-appengine > com.google.appengine:appengine-api-1.0-sdk' exist in all 9 dependency paths. Example path: com.google.cloud:google-cloud-core-http:1.59.0 / com.google.http-client:google-http-client-appengine:1.27.0 / com.google.appengine:appengine-api-1.0-sdk:1.9.64